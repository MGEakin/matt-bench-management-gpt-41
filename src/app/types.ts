import { z } from "zod";

export const EmployeeSchema = z.object({
  id: z.number().optional(),
  name: z.string(),
  title: z.string(),
  studio: z.string(),
  practice: z.string(),
  region: z.string(),
  location: z.string(),
  skill_level: z.string(),
  current_assignment: z.string().optional(),
});

export type Employee = z.infer<typeof EmployeeSchema>;
