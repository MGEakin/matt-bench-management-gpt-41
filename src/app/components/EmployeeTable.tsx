import React from "react";
import Link from "next/link";
import { Employee } from "../../types/employee";

interface EmployeeTableProps {
  employees: Employee[];
  onEdit: (employee: Employee) => void;
  onDelete: (id: number) => void;
}

const EmployeeTable: React.FC<EmployeeTableProps> = ({
  employees,
  onEdit,
  onDelete,
}) => (
  <div className="overflow-x-auto w-full [container-type: inline-size]">
    <table className="min-w-full border border-gray-200 rounded-lg shadow-sm">
      <thead className="bg-gray-100 dark:bg-gray-800">
        <tr>
          <th className="px-4 py-2">Name</th>
          <th className="px-4 py-2">Title</th>
          <th className="px-4 py-2">Studio</th>
          <th className="px-4 py-2">Practice</th>
          <th className="px-4 py-2">Region</th>
          <th className="px-4 py-2">Location</th>
          <th className="px-4 py-2">Skill Level</th>
          <th className="px-4 py-2">Current Assignment</th>
          <th className="px-4 py-2">Actions</th>
        </tr>
      </thead>
      <tbody>
        {employees.length === 0 ? (
          <tr>
            <td colSpan={9} className="px-4 py-8 text-center text-gray-500">
              No employees found
            </td>
          </tr>
        ) : (
          employees.map((emp) => (
            <tr key={emp.id} className="border-t">
              <td className="px-4 py-2 text-xs md:text-sm lg:text-base">
                <Link
                  href={`/employee/${emp.id}`}
                  className="text-blue-700 underline hover:text-blue-900"
                >
                  {emp.name}
                </Link>
              </td>
              <td className="px-4 py-2 text-xs md:text-sm lg:text-base">
                {emp.title}
              </td>
              <td className="px-4 py-2 text-xs md:text-sm lg:text-base">
                <Link
                  href={`/studio/${encodeURIComponent(emp.studio)}`}
                  className="text-blue-700 underline hover:text-blue-900"
                >
                  {emp.studio}
                </Link>
              </td>
              <td className="px-4 py-2 text-xs md:text-sm lg:text-base">
                <Link
                  href={`/practice/${encodeURIComponent(emp.practice)}`}
                  className="text-blue-700 underline hover:text-blue-900"
                >
                  {emp.practice}
                </Link>
              </td>
              <td className="px-4 py-2 text-xs md:text-sm lg:text-base">
                {emp.region}
              </td>
              <td className="px-4 py-2 text-xs md:text-sm lg:text-base">
                {emp.location}
              </td>
              <td className="px-4 py-2 text-xs md:text-sm lg:text-base">
                {emp.skill_level}
              </td>
              <td className="px-4 py-2 text-xs md:text-sm lg:text-base">
                {emp.current_assignment || "-"}
              </td>
              <td className="px-4 py-2 flex gap-2">
                <button
                  className="text-blue-600 hover:underline"
                  onClick={() => onEdit(emp)}
                >
                  Edit
                </button>
                <button
                  className="text-red-600 hover:underline"
                  onClick={() => emp.id !== undefined && onDelete(emp.id)}
                >
                  Delete
                </button>
              </td>
            </tr>
          ))
        )}
      </tbody>
    </table>
  </div>
);

export default EmployeeTable;
