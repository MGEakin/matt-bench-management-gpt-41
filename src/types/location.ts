import { z } from "zod";

export const LocationSchema = z.object({
  id: z.number().optional(),
  name: z.string(),
});

export type Location = z.infer<typeof LocationSchema>;
