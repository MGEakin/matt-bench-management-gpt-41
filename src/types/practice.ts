import { z } from "zod";

export const PracticeSchema = z.object({
  id: z.number(),
  name: z.string(),
});

export type Practice = z.infer<typeof PracticeSchema>;
