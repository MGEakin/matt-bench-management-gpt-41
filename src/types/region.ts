import { z } from "zod";

export const RegionSchema = z.object({
  id: z.number().optional(),
  name: z.string(),
});

export type Region = z.infer<typeof RegionSchema>;
