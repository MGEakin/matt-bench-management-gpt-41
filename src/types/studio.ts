import { z } from "zod";

export const StudioSchema = z.object({
  id: z.number().optional(),
  name: z.string(),
});

export type Studio = z.infer<typeof StudioSchema>;
