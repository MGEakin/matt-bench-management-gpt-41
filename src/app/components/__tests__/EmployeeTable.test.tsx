import React from "react";
import { describe, it, expect, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import EmployeeTable from "../EmployeeTable";
import { Employee } from "../../types";

import "@testing-library/jest-dom";
import { create } from "react-test-renderer";

describe("EmployeeTable", () => {
  const employees: Employee[] = [
    {
      id: 1,
      name: "Alice Smith",
      title: "QA Lead",
      studio: "Studio A",
      practice: "Testing",
      region: "Americas",
      location: "New York",
      skill_level: "Senior",
      current_assignment: "Project X",
    },
  ];

  it("renders employee rows", () => {
    render(
      <EmployeeTable
        employees={employees}
        onEdit={() => {}}
        onDelete={() => {}}
      />,
    );
    expect(screen.getByText("Alice Smith")).toBeInTheDocument();
    expect(screen.getByText("QA Lead")).toBeInTheDocument();
  });

  it("renders empty state when no employees", () => {
    render(
      <EmployeeTable employees={[]} onEdit={() => {}} onDelete={() => {}} />,
    );
    expect(screen.getByText(/No employees found/i)).toBeInTheDocument();
  });

  it("calls onEdit when Edit button is clicked", () => {
    const onEdit = vi.fn();
    render(
      <EmployeeTable
        employees={employees}
        onEdit={onEdit}
        onDelete={() => {}}
      />,
    );
    const editButton = screen.getByRole("button", { name: /Edit/i });
    editButton.click();
    expect(onEdit).toHaveBeenCalledWith(employees[0]);
  });

  it("calls onDelete when Delete button is clicked", () => {
    const onDelete = vi.fn();
    render(
      <EmployeeTable
        employees={employees}
        onEdit={() => {}}
        onDelete={onDelete}
      />,
    );
    const deleteButton = screen.getByRole("button", { name: /Delete/i });
    deleteButton.click();
    expect(onDelete).toHaveBeenCalledWith(employees[0].id);
  });

  it("matches snapshot", () => {
    const tree = create(
      <EmployeeTable
        employees={employees}
        onEdit={() => {}}
        onDelete={() => {}}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has accessible table", () => {
    render(
      <EmployeeTable
        employees={employees}
        onEdit={() => {}}
        onDelete={() => {}}
      />,
    );
    expect(screen.getByRole("table")).toBeInTheDocument();
  });
});
