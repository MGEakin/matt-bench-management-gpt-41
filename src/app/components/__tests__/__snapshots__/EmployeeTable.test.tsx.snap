// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmployeeTable > matches snapshot 1`] = `null`;
